src/morphology/english/getNounForms.js

```diff
@@ -0,0 +1,88 @@
+// "use strict";
+const irregularNouns = require( "./irregularNouns.js" );
+const singularizeRegex = require( "./regexNoun.js" ).singularizeRegex;
+const pluralizeRegex = require( "./regexNoun.js" ).pluralizeRegex;
+const hispanicRegex = require( "./regexNoun.js" ).hispanicRegex;
+
+const isUndefined = require( "lodash/isUndefined.js" );
+const unique = require( "lodash/uniqBy" );
+
+const checkIrregulars = function( word ) {
+	let irregulars;
+
+	irregularNouns.forEach( function( paradigm ) {
+		paradigm.forEach( function( wordInParadigm ) {
+			if ( wordInParadigm === word ) {
+				irregulars = paradigm;
+			}
+		} );
+	} );
+	return irregulars;
+};
+
+const checkHispanic = function( word ) {
+	for ( let i = 0; i < hispanicRegex.length; i++ ) {
+		if ( hispanicRegex[ i ].reg.test( word ) === true ) {
+			return [
+				word.replace( hispanicRegex[ i ].reg, hispanicRegex[ i ].repl1 ),
+				word.replace( hispanicRegex[ i ].reg, hispanicRegex[ i ].repl2 ),
+			];
+		}
+	}
+};
+
+const singularize = function( word ) {
+	for ( let i = 0; i < singularizeRegex.length; i++ ) {
+		if ( singularizeRegex[ i ].reg.test( word ) === true ) {
+			return word.replace( singularizeRegex[ i ].reg, singularizeRegex[ i ].repl );
+		}
+	}
+};
+
+const pluralize = function( word ) {
+	for ( let i = 0; i < pluralizeRegex.length; i++ ) {
+		if ( pluralizeRegex[ i ].reg.test( word ) === true ) {
+			return word.replace( pluralizeRegex[ i ].reg, pluralizeRegex[ i ].repl );
+		}
+	}
+};
+
+
+const getNounForms = function( word ) {
+	let forms = [];
+	let isIrregular = false;
+
+	const irregular = checkIrregulars( word );
+	if ( ! isUndefined( irregular ) ) {
+		forms = forms.concat( irregular );
+		isIrregular = true;
```

Comment:
Instead of setting `isIrregular` to true and then checking whether it's true, I think you can just declare the `return unique( forms );` statement at this position.

Reply:
Good point.